pimd/pim_cmd_common.c

```diff
@@ -653,3 +673,1302 @@ int pim_process_no_rp_plist_cmd(struct vty *vty, const char *rp_str,
+{
+	if (up->flags & PIM_UPSTREAM_FLAG_MASK_DR_JOIN_DESIRED)
+		json_object_boolean_true_add(json, "drJoinDesired");
+
+	if (up->flags & PIM_UPSTREAM_FLAG_MASK_DR_JOIN_DESIRED_UPDATED)
+		json_object_boolean_true_add(json, "drJoinDesiredUpdated");
+
+	if (up->flags & PIM_UPSTREAM_FLAG_MASK_FHR)
+		json_object_boolean_true_add(json, "firstHopRouter");
+
+	if (up->flags & PIM_UPSTREAM_FLAG_MASK_SRC_IGMP)
+		json_object_boolean_true_add(json, "sourceIgmp");
+
+	if (up->flags & PIM_UPSTREAM_FLAG_MASK_SRC_PIM)
+		json_object_boolean_true_add(json, "sourcePim");
+
+	if (up->flags & PIM_UPSTREAM_FLAG_MASK_SRC_STREAM)
+		json_object_boolean_true_add(json, "sourceStream");
+
+	/* XXX: need to print ths flag in the plain text display as well */
+	if (up->flags & PIM_UPSTREAM_FLAG_MASK_SRC_MSDP)
+		json_object_boolean_true_add(json, "sourceMsdp");
+
+	if (up->flags & PIM_UPSTREAM_FLAG_MASK_SEND_SG_RPT_PRUNE)
+		json_object_boolean_true_add(json, "sendSGRptPrune");
+
+	if (up->flags & PIM_UPSTREAM_FLAG_MASK_SRC_LHR)
+		json_object_boolean_true_add(json, "lastHopRouter");
+
+	if (up->flags & PIM_UPSTREAM_FLAG_MASK_DISABLE_KAT_EXPIRY)
+		json_object_boolean_true_add(json, "disableKATExpiry");
+
+	if (up->flags & PIM_UPSTREAM_FLAG_MASK_STATIC_IIF)
+		json_object_boolean_true_add(json, "staticIncomingInterface");
+
+	if (up->flags & PIM_UPSTREAM_FLAG_MASK_ALLOW_IIF_IN_OIL)
+		json_object_boolean_true_add(json,
+					     "allowIncomingInterfaceinOil");
+
+	if (up->flags & PIM_UPSTREAM_FLAG_MASK_NO_PIMREG_DATA)
+		json_object_boolean_true_add(json, "noPimRegistrationData");
+
+	if (up->flags & PIM_UPSTREAM_FLAG_MASK_FORCE_PIMREG)
+		json_object_boolean_true_add(json, "forcePimRegistration");
+
+	if (up->flags & PIM_UPSTREAM_FLAG_MASK_SRC_VXLAN_ORIG)
+		json_object_boolean_true_add(json, "sourceVxlanOrigination");
+
+	if (up->flags & PIM_UPSTREAM_FLAG_MASK_SRC_VXLAN_TERM)
+		json_object_boolean_true_add(json, "sourceVxlanTermination");
+
+	if (up->flags & PIM_UPSTREAM_FLAG_MASK_MLAG_VXLAN)
+		json_object_boolean_true_add(json, "mlagVxlan");
+
+	if (up->flags & PIM_UPSTREAM_FLAG_MASK_MLAG_NON_DF)
+		json_object_boolean_true_add(json,
+					     "mlagNonDesignatedForwarder");
+}
+
+static void pim_show_join_helper(struct vty *vty, struct pim_interface *pim_ifp,
+				 struct pim_ifchannel *ch, json_object *json,
+				 time_t now)
+{
+	json_object *json_iface = NULL;
+	json_object *json_row = NULL;
+	json_object *json_grp = NULL;
+	pim_addr ifaddr;
+	char uptime[10];
+	char expire[10];
+	char prune[10];
+
+	ifaddr = pim_ifp->primary_address;
+
+	pim_time_uptime_begin(uptime, sizeof(uptime), now, ch->ifjoin_creation);
+	pim_time_timer_to_mmss(expire, sizeof(expire),
+			       ch->t_ifjoin_expiry_timer);
+	pim_time_timer_to_mmss(prune, sizeof(prune),
+			       ch->t_ifjoin_prune_pending_timer);
+
+	if (json) {
+		char ch_grp_str[PIM_ADDRSTRLEN];
+		char ch_src_str[PIM_ADDRSTRLEN];
+
+		snprintfrr(ch_grp_str, sizeof(ch_grp_str), "%pPAs",
```

Comment:
move formatting into `json_object_string_addf` to remove extra step with `ch_grp_str`/`ch_src_str`

Reply:
Done